### PR TITLE
Modified description to indicate an optional step.

### PR DESCRIPTION
**Pull request by corneil**
Added TODO for potential changes to move all Optional steps further down where appropriate.

### CODE REVIEW

File: content/documentation/pages/5-feature-guides/2-streams/10-function-composition.md [hunk 1]
```diff
@@ -184,19 +184,21 @@ We can now deploy a new stream by using the `http-transform` application that in
 stream create helloComposed --definition "http-transformer --server.port=9001 | log"
 ```
 
-To define the order in which the functions are run, we have to use the `spring.cloud.stream.function.definition` property to define the function definition.
+[//]: # 'TODO clarify that this is optional. Move optional to the end'
```
Comment (onobc) [inline content/documentation/pages/5-feature-guides/2-streams/10-function-composition.md:187]: Yeh, good call on verifying that it is "in fact" optional. Once you do this we can merge this in. Thanks for updating.

Comment (dturanski) [reply content/documentation/pages/5-feature-guides/2-streams/10-function-composition.md:187]: should be `spring.cloud.function.definition`

File: content/documentation/pages/5-feature-guides/2-streams/10-function-composition.md [hunk 2]
```diff
@@ -184,19 +184,21 @@ We can now deploy a new stream by using the `http-transform` application that in
...
 In this case, it is as follows:
 
 ```
-stream deploy helloComposed --properties "app.http-transformer.spring.cloud.stream.function.definition=upper|concat"
+stream deploy helloComposed --properties "app.http-transformer.spring.cloud.stream.function.definition=concat|upper"
 
 ```
 
-<!-- TODO why not specify function.definition in the stream definition? -->
+[//]: # 'TODO why not specify function.definition in the stream definition?'
```
Comment (dturanski) [inline content/documentation/pages/5-feature-guides/2-streams/10-function-composition.md:199]: The stream dsl parser consumes `I` and gets confused. You may try escaping but `,` is also supported. https://github.com/spring-cloud/stream-applications/blob/main/applications/stream-applications-core/stream-applications-composite-function-support/src/main/java/org/springframework/cloud/stream/app/composite/function/common/FunctionBindingEnvironmentPostProcessor.java#L98

Comment (corneil) [reply content/documentation/pages/5-feature-guides/2-streams/10-function-composition.md]: > The stream dsl parser consumes `|` and gets confused. You may try escaping but `,` is also supported. https://github.com/spring-cloud/stream-applications/blob/main/applications/stream-applications-core/stream-applications-composite-function-support/src/main/java/org/springframework/cloud/stream/app/composite/function/common/FunctionBindingEnvironmentPostProcessor.java#L98
@dturanski 
This specific method is used to create a channel name for that function.

Do we have any examples where the | in a function definition is causing problems?

**Pull request merged**